waspc/src/Wasp/Generator/WebAppGenerator/RouterGenerator.hs

```diff
@@ -59,35 +60,39 @@ instance ToJSON PageTemplateData where
         "importFrom" .= _importFrom pageTD
       ]
 
-generateRouter :: AppSpec -> FileDraft
-generateRouter spec =
-  C.mkTmplFdWithDstAndData
-    (asTmplFile $ [reldir|src|] </> routerPath)
-    targetPath
-    (Just $ toJSON templateData)
+generateRouter :: AppSpec -> Generator FileDraft
+generateRouter spec = do
+  templateData <- createRouterTemplateData spec
+  return $
+    C.mkTmplFdWithDstAndData
+      (asTmplFile $ [reldir|src|] </> routerPath)
+      targetPath
+      (Just $ toJSON templateData)
   where
     routerPath = [relfile|router.js|]
-    templateData = createRouterTemplateData spec
     targetPath = C.webAppSrcDirInWebAppRootDir </> asWebAppSrcFile routerPath
 
-createRouterTemplateData :: AppSpec -> RouterTemplateData
-createRouterTemplateData spec =
-  RouterTemplateData
-    { _routes = routes,
-      _pagesToImport = pages,
-      _isAuthEnabled = AS.isAuthEnabled spec
-    }
+createRouterTemplateData :: AppSpec -> Generator RouterTemplateData
+createRouterTemplateData spec = do
+  routes <- mapM (createRouteTemplateData spec) $ AS.getRoutes spec
+  return $
+    RouterTemplateData
+      { _routes = routes,
+        _pagesToImport = pages,
+        _isAuthEnabled = AS.isAuthEnabled spec
+      }
   where
-    routes = map (createRouteTemplateData spec) $ AS.getRoutes spec
     pages = map createPageTemplateData $ AS.getPages spec
 
-createRouteTemplateData :: AppSpec -> (String, AS.Route.Route) -> RouteTemplateData
+createRouteTemplateData :: AppSpec -> (String, AS.Route.Route) -> Generator RouteTemplateData
```

Comment:
I believe you actually don't need to use `Generator` here any more? It can just return `RouteTemplateData`, since this is not really a generator function, but a helper function?

Reply:
Good point, will remove.